sklearn/metrics/tests/test_classification.py

```diff
@@ -2246,10 +2246,24 @@ def test_brier_score_loss():
     assert_almost_equal(brier_score_loss([-1], [0.4]), 0.16)
     assert_almost_equal(brier_score_loss([0], [0.4]), 0.16)
     assert_almost_equal(brier_score_loss([1], [0.4]), 0.36)
-    assert_almost_equal(
-        brier_score_loss(['foo'], [0.4], pos_label='bar'), 0.16)
-    assert_almost_equal(
-        brier_score_loss(['foo'], [0.4], pos_label='foo'), 0.36)
+
+    # make sure the positive class is correctly inferred
+    y_true = np.array([0, 1, 1, 0])
+    y_pred = np.array([0.8, 0.6, 0.4, 0.2])
+    score1 = brier_score_loss(y_true, y_pred, pos_label=1)
+    score2 = brier_score_loss(y_true, y_pred)
+    assert score1 == pytest.approx(score2)
+    y_true = np.array(["neg", "pos", "pos", "neg"])
+    # raise error when y_true contains strings and pos_label is not specified
+    with pytest.raises(ValueError, match="pos_label must be specified"):
+        brier_score_loss(y_true, y_pred)
+    score2 = brier_score_loss(y_true, y_pred, pos_label="pos")
+    assert score1 == pytest.approx(score2)
+
+    # positive class if correctly inferred an object array with all ints
+    y_pred_num_obj = np.array([0, 1, 1, 0], dtype=object)
```

Comment:
This test case was added which was enabled by https://github.com/scikit-learn/scikit-learn/pull/15412#discussion_r415834438 and with this https://github.com/scikit-learn/scikit-learn/pull/15412/commits/fb199bd680431fe6c6e4b5c7c2834299ac9e7c24 diff